source/slang/slang-emit-glsl.cpp

```diff
@@ -310,7 +334,16 @@ void GLSLSourceEmitter::_emitGLSLParameterGroup(IRGlobalParam* varDecl, IRUnifor
     or IRGLSLShaderStorageBufferType which is read write.
     */
 
-    _emitGLSLLayoutQualifier(LayoutResourceKind::DescriptorTableSlot, &containerChain);
+    {
+        // TODO(JS):
+        // The *assumption* here is that we only need to detect the LayoutResourceKind as used on the initial containerChain.varLayout
+        // rather than search up the chain. 
```

Comment:
I believe this is correct, yes. The kinds of resources consumed by the leaf variable will only depend on its type, and its type is determined by the layout of the last `varLayout` in the chain.

(The rest of the "chain" is used to summing up all the other relevant offsets)